app/src/main/java/com/github/multimatum_team/multimatum/DeadlineNotification.kt

```diff
@@ -21,35 +22,83 @@ class DeadlineNotification {
     Create a notification channel for reminder notifications
     Creating an existing notification channel with its original values performs no operation,
     so it's safe to call this code when starting an app.
-    */
-    fun createNotificationChannel(context: Context){
-        val channelName :CharSequence = "reminders channel"
+     */
+    fun createNotificationChannel(context: Context) {
+        val channelName: CharSequence = "reminders channel"
         val description = "channel for reminders notifications"
-        val channel = NotificationChannel("remindersChannel", channelName, NotificationManager.IMPORTANCE_DEFAULT)
-        val notificationManager = context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val channel = NotificationChannel(
+            "remindersChannel",
+            channelName,
+            NotificationManager.IMPORTANCE_DEFAULT
+        )
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
 
-        channel.description=description
+        channel.description = description
 
         notificationManager.createNotificationChannel(channel)
     }
 
     /**
-    Set a notification that will be triggered in a given time in ms.
-    you can pass a title/description and Id in parameter
-    */
-    fun setNotification(deadline: Deadline, context: Context, triggerTime: Long){
-        alarmManager = context.getSystemService(AppCompatActivity.ALARM_SERVICE) as AlarmManager  //this get an service instance of AlarmManager
-        val intent = Intent(context, ReminderBroadcastReceiver::class.java) //this create an intent of broadcast receiver
-        //Adding extra parameter that will be used in the broadcase receiver to create the notification
+     * Set a notification for a deadline in a given time before it is due.
+     *
+     * For example, to set an alarm (=notif) 1 hours before the deadline is due, use "setNotification(Pair(deadline_id, deadline), context, Duration.of(1, ChronoUnit.HOURS).toMillis())
```

Comment:
You could separate this comment in two lines

Reply:
Done